### DISCUSSION TITLE
What do you use for i18n in Remix?

### DISCUSSION THREAD

**Discussion by knpwrs** [remix-run/remix#2724] [Q&A] [answered]
I'm personally a fan of https://github.com/lingui/js-lingui, though it's heavily tied in to the Babel ecosystem. I could probably set something up through https://github.com/aiji42/remix-esbuild-override, but this feels... fragile. For those who have experience with Remix and i18n, what do you use?

Worth nothing, what I like about Lingui is that you can use standard React components to write your strings, and it'll extract ICU-formatted strings for translation, all while supporting rich text and component interpolation.

**Comment by sergiodxa** [comment, marked as answer]
I never used Lingui but I worked with i18next before and I built an integration package for Remix on  https://github.com/sergiodxa/remix-i18next, I'm working now on a new major version with a better integration history with the core i18next lib.

**Reply by knpwrs** [reply]
It looks like i18next has improved a lot since the last time I'd used it. I'll have to give it another go.
